### PR TITLE
added yandex 1B subset dataset generator

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: create_text_to_image_ds.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+from ann_benchmarks.algorithms.bruteforce import BruteForceBLAS
+import struct
+import numpy as np
+import click
+import h5py
+from joblib import Parallel, delayed
+import multiprocessing
+
+def read_fbin(filename, start_idx=0, chunk_size=None):
```
Comment (GuyAv46) [inline create_text_to_image_ds.py:9]: consider having `float` and `int` as parameters?

File: create_text_to_image_ds.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    """
+    with open(filename, "rb") as f:
+        nvecs, dim = np.fromfile(f, count=2, dtype=np.int32)
+        nvecs = (nvecs - start_idx) if chunk_size is None else chunk_size
+        arr = np.fromfile(f, count=nvecs * dim, dtype=np.int32, 
+                          offset=start_idx * 4 * dim)
+    return arr.reshape(nvecs, dim)
+ 
+ 
+def write_fbin(filename, vecs):
```
Comment (GuyAv46) [inline create_text_to_image_ds.py:45]: same as for the read commands

File: create_text_to_image_ds.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+def write_output(train, test, fn, distance, point_type='float', count=100):
+    n = 0
+    f = h5py.File(fn, 'w')
+    f.attrs['type'] = 'dense'
+    f.attrs['distance'] = distance
+    f.attrs['dimension'] = len(train[0])
+    f.attrs['point_type'] = point_type
+    print('train size: %9d * %4d' % train.shape)
+    print('test size:  %9d * %4d' % test.shape)
+    f.create_dataset('train', (len(train), len(
```
Comment (GuyAv46) [inline create_text_to_image_ds.py:94]: Is this an ann_benchmark function?

File: create_text_to_image_ds.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+@click.option('--distance', default='angular', help='distance metric.')
+@click.option('--test_set', required=True, type=str)
+@click.option('--train_set', required=True, type=str)
+def create_ds(size, distance, test_set, train_set):
+    test_set = read_fbin(test_set)
+    train_set= read_fbin(train_set, chunk_size=size*1000000)
+    write_output(train=train_set, test=test_set, fn=f'Text-to-Image-{size}M.hd5f', distance=distance, point_type='float', count=100)
+
+if __name__ == "__main__":
+    create_ds()
```
Comment (GuyAv46) [inline create_text_to_image_ds.py]: ... :)

File: create_text_to_image_ds.py (hunk 1 above)
Comment (DvirDukhan) [reply create_text_to_image_ds.py:9]: why?

File: create_text_to_image_ds.py (hunk 3 above)
Comment (DvirDukhan) [reply create_text_to_image_ds.py:94]: yes

File: create_text_to_image_ds.py (hunk 4 above)
Comment (DvirDukhan) [reply create_text_to_image_ds.py]: didn't get this comment :)

File: create_text_to_image_ds.py (hunk 1 above)
Comment (GuyAv46) [reply create_text_to_image_ds.py:9]: Im not sure why we need the two functions and not only float, but it is code duplication beside the type, and maybe in the future we will want to support more types

File: create_text_to_image_ds.py (hunk 4 above)
Comment (GuyAv46) [reply create_text_to_image_ds.py]: missing new line

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**